src/gui/mainwindow.cpp

```diff
@@ -147,8 +147,14 @@ bool MainWindow::createProjectConfig(const QString& name)
 {
     if (!closeProjectConfig())
         return false;
-
-    projectConfig = std::make_unique<ProjectConfig>(this);
+  if (configMgr.get() == nullptr) {
+        configMgr = std::make_unique<ConfigMgr>();
+        if (configMgr->init() == false) {
+            cds_error("ConfigMgr not initialized correctly.");
+            return false;
+        }
+    }
+    projectConfig = std::make_unique<ProjectConfig>(this, configMgr.get());
```

Comment:
Passing anything by pointer makes ownership unclear. Furthermore you are passing here pointer already managed by unique_ptr and later on you are assigning it to other unique_ptrs.